source/validate.cpp

```diff
@@ -215,6 +245,16 @@ spv_result_t ValidateBinaryUsingContextAndValidationState(
            << "Invalid SPIR-V header.";
   }
 
+
+  // Look for OpExtension instructions and register extensions.
+  // Diagnostics if any will be produced in the next pass (ProcessInstruction).
+  if (auto error = spvBinaryParse(&context, vstate, words, num_words,
+                                  setHeader, ProcessExtensions,
```

Comment:
It would be clearer to use nullptr instead of setHeader here.  You only care about processing instructions, not the header.